src/ota_proxy/ota_cache.py

```diff
@@ -641,3 +671,29 @@ async def retrieve_file(
         return await self._retrieve_file_by_downloading(
             raw_url, headers=headers_from_client
         )
+
+    def _background_update_metrics(self):
+        """Constantly updates shared memory with metrics data.
+
+        This method runs in a background thread and periodically writes
+        the current metrics data to shared memory using the SharedOTAClientMetricsWriter.
+        """
+        while self._metrics_thread_running:
+            try:
+                if self._shm_metrics_writer:
+                    # Write current metrics data to shared memory
+                    self._shm_metrics_writer.write_msg(self._metrics_data)
+                    logger.debug("Metrics data updated to shared memory")
+
+                # Wait for the specified interval before next update
+                time.sleep(cfg.METRICS_UPDATE_INTERVAL)
+            except Exception as e:
+                logger.error(f"Error updating metrics to shared memory: {e}")
+                # Continue running even if there's an error
+                time.sleep(cfg.METRICS_UPDATE_INTERVAL)
+
+        try:
+            if self._shm_metrics_writer:
+                self._shm_metrics_writer.write_msg(self._metrics_data)
+        except Exception as e:
+            logger.error(f"Error updating metrics to shared memory: {e}")
```

Comment:
background thread to write the cache metrics to shared memory, current interval is 5 sec.